src/primitives/block.h

```diff
@@ -78,7 +78,7 @@ class CBlock : public CBlockHeader
     std::vector<CTransaction> vtx;
 
     // memory only
-    mutable std::vector<uint256> vMerkleTree;
+    mutable bool fChecked;
```

Comment:
Fully agree, I actually started implementing that as a follow-up, but didn't want to interfere with other refactorings.